screen/loans/src/main/java/com/ivy/loans/loan/LoanScreenState.kt

```diff
@@ -9,12 +9,16 @@ import java.time.Instant
 data class LoanScreenState(
     val baseCurrency: String,
     val loans: ImmutableList<DisplayLoan>,
+    val completedLoans: ImmutableList<DisplayLoan>,
+    val pendingLoans: ImmutableList<DisplayLoan>,
     val accounts: ImmutableList<Account>,
     val selectedAccount: Account?,
     val loanModalData: LoanModalData?,
     val reorderModalVisible: Boolean,
     val totalOweAmount: String,
     val totalOwedAmount: String,
     val paidOffLoanVisibility: Boolean,
-    val dateTime: Instant
-)
+    val tabularLoanMode: Boolean,
+    val dateTime: Instant,
+    val selectedTab: LoanTab
+)
```

Comment:
Make the state a sealed interface with two cases - tabular and regulad

Reply:
Done

Reply:
@shamim-emon I meant:
```kotlin
sealed interface LoanScreenViewState {
  // common fields

  data class Regular(
    val loans: ImmutableList<Loan>,
    override val for all common
   ) : LoanScreenViewState
   
   data class Tabular(
     val completedLoans: ImmutableList<DisplayLoan>,
    val pendingLoans: ImmutableList<DisplayLoan>,
   ) : LoanScreenViewState
}
```

This data modeling fixes the issue where "loans" is useless for the tabular UI and completed + pending is useless for the regular mode. See the data modeling guidelines in Ivy - this eliminates impossible cases and better represent reality. 

However, we're good to merge it. Tbh, your UI is better than the original so we might just get rid of the setting and maks the tabular UI the new Loans tab UI. I'll merge the PR, if you have time feel free to make the tabular UI the only Loans UI and cleanup the regular UI code

Reply:
@ILIYANGERMANOV I have taken up the task. Will get rid of the old loan UI & relevant code.